sogs/routes/auth.py

```diff
@@ -135,6 +135,36 @@ def require_mod(room, *, admin=False):
         )
 
 
+def accessible_required(f):
+    """Decorator for an endpoint that requires a user have accessible or read permission in the
+    given room.  The function must take a `room` argument by name, as is typically used with flask
+    endpoints with a `<Room:room>` argument."""
+
+    @wraps(f)
+    def required_accessible_wrapper(*args, room, **kwargs):
+        if not room.check_accessible(g.user):
+            abort_with_reason(http.FORBIDDEN, "This room is not accessible", warn=False)
```

Comment:
this can leak info between rooms not existing and rooms not being accessible.

Reply:
Yes, but does that actually matter?

Reply:
yes, the point of this flag was to prevent this type of info leak.

Reply:
Updated to 404 instead.